src/diffusers/models/modeling_utils.py

```diff
@@ -562,6 +623,41 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             **kwargs,
         )
 
+        # Determine if we're loading from a directory of sharded checkpoints.
+        is_sharded = False
+        is_local = os.path.isdir(pretrained_model_name_or_path)
+        if is_local:
+            index_file = Path(
+                pretrained_model_name_or_path,
+                subfolder,
+                _add_variant(SAFE_WEIGHTS_INDEX_NAME if use_safetensors else WEIGHTS_INDEX_NAME, variant),
+            ).as_posix()
+            if os.path.isfile(index_file):
+                is_sharded = True
+        else:
+            index_file = Path(
+                subfolder, _add_variant(SAFE_WEIGHTS_INDEX_NAME if use_safetensors else WEIGHTS_INDEX_NAME, variant)
+            ).as_posix()
```

Comment:
I guess here `as_posix()` makes sense because we are passing a relative path in repo (which is always a unix-like path) in opposition to when using local paths (which can be unix-like or windows-like paths) :+1: